spec/lib/video_info/providers/youtube_api_spec.rb

```diff
@@ -0,0 +1,392 @@
+    subject { VideoInfo.new('http://www.youtube.com/user/EducatorVids3?v=VeasFckfMHY') }
+
+    describe '#provider' do
+      subject { super().provider }
+      it { is_expected.to eq 'YouTube' }
+    end
+
+    describe '#video_id' do
+      subject { super().video_id }
+      it { is_expected.to eq 'VeasFckfMHY' }
+    end
+
+    describe '#url' do
+      subject { super().url }
+      it { is_expected.to eq 'http://www.youtube.com/user/EducatorVids3?v=VeasFckfMHY' }
+    end
+  end
+
+  context "with video VeasFckfMHY after params", :vcr do
+    subject { VideoInfo.new('http://www.youtube.com/watch?feature=player_profilepage&v=VeasFckfMHY') }
+
+    describe '#provider' do
+      subject { super().provider }
+      it { is_expected.to eq 'YouTube' }
+    end
+
+    describe '#video_id' do
+      subject { super().video_id }
+      it { is_expected.to eq 'VeasFckfMHY' }
+    end
+
+    describe '#url' do
+      subject { super().url }
+      it { is_expected.to eq 'http://www.youtube.com/watch?feature=player_profilepage&v=VeasFckfMHY' }
+    end
+  end
+
+  context "with video VeasFckfMHY in path", :vcr do
+    subject { VideoInfo.new('http://www.youtube.com/v/VeasFckfMHY') }
+
+    describe '#provider' do
+      subject { super().provider }
+      it { is_expected.to eq 'YouTube' }
+    end
+
+    describe '#video_id' do
+      subject { super().video_id }
+      it { is_expected.to eq 'VeasFckfMHY' }
+    end
+  end
+
+  context "with video VeasFckfMHY in e path", :vcr do
+    subject { VideoInfo.new('http://www.youtube.com/e/VeasFckfMHY') }
+
+    describe '#provider' do
+      subject { super().provider }
+      it { is_expected.to eq 'YouTube' }
+    end
+
+    describe '#video_id' do
+      subject { super().video_id }
+      it { is_expected.to eq 'VeasFckfMHY' }
+    end
+  end
+
+  context "with video VeasFckfMHY in embed path", :vcr do
+    subject { VideoInfo.new('http://www.youtube.com/embed/VeasFckfMHY') }
+
+    describe '#provider' do
+      subject { super().provider }
+      it { is_expected.to eq 'YouTube' }
+    end
+
+    describe '#video_id' do
+      subject { super().video_id }
+      it { is_expected.to eq 'VeasFckfMHY' }
+    end
+  end
+
+  context "with video JM9NgvjjVng in youtu.be url", :vcr do
+    subject { VideoInfo.new('http://youtu.be/JM9NgvjjVng') }
+
+    describe '#provider' do
+      subject { super().provider }
+      it { is_expected.to eq 'YouTube' }
+    end
+
+    describe '#video_id' do
+      subject { super().video_id }
+      it { is_expected.to eq 'JM9NgvjjVng' }
+    end
+  end
+
+  context 'without http or www', :vcr do
+    subject { VideoInfo.new('youtu.be/JM9NgvjjVng') }
+
+    describe '#provider' do
+      subject { super().provider }
+      it { is_expected.to eq('YouTube') }
+    end
+
+    describe '#video_id' do
+      subject { super().video_id }
+      it { is_expected.to eq 'JM9NgvjjVng' }
+    end
+  end
+
+  context "with video url in text", :vcr do
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.